pkg/controller/hibernation/aws_actuator.go

```diff
@@ -0,0 +1,150 @@
+package hibernation
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awsclient "github.com/openshift/hive/pkg/awsclient"
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+// AWS machine states
+const (
+	instancePending      = 0
+	instanceRunning      = 16
+	instanceShuttingDown = 32
+	instanceTerminated   = 48
+	instanceStopping     = 64
+	instanceStopped      = 80
+)
+
+func init() {
+	RegisterActuator(&awsActuator{})
+}
+
+type awsActuator struct {
+}
+
+// CanHandle returns true if the actuator can handle a particular ClusterDeployment
+func (a *awsActuator) CanHandle(cd *hivev1.ClusterDeployment) bool {
+	return cd.Spec.Platform.AWS != nil
+}
+
+// StopMachines will start machines belonging to the given ClusterDeployment
+func (a *awsActuator) StopMachines(logger log.FieldLogger, cd *hivev1.ClusterDeployment, c client.Client) error {
+	logger.Infof("AWS: stopping machines")
```

Comment:
Instead of "AWS:" would it be better to have a field for something like "cloud=aws" ?

That's the pattern I've seen in other places. It allows us to search for this specific field in logs.